src/libraries/Microsoft.Extensions.Logging/src/Logger.cs

```diff
@@ -178,12 +178,30 @@ public List<LoggerProviderDebugView> Providers
                     {
                         LoggerInformation loggerInfo = logger.Loggers[i];
                         string providerName = ProviderAliasUtilities.GetAlias(loggerInfo.ProviderType) ?? loggerInfo.ProviderType.Name;
-                        MessageLogger? messageLogger = logger.MessageLoggers?.FirstOrDefault(messageLogger => messageLogger.Logger == loggerInfo.Logger);
+                        MessageLogger? messageLogger = FirstOrNull(logger.MessageLoggers, loggerInfo.Logger);
 
                         providers.Add(new LoggerProviderDebugView(providerName, messageLogger));
                     }
 
                     return providers;
+
+                    static MessageLogger? FirstOrNull(MessageLogger[]? messageLoggers, ILogger logger)
```

Comment:
```suggestion
                    // Find message logger or return null if there is no match. FirstOrDefault isn't used because MessageLogger is a struct.
                    static MessageLogger? FirstOrNull(MessageLogger[]? messageLoggers, ILogger logger)
```